api/v1beta1/openstackdataplanenodeset_webhook.go

```diff
@@ -87,8 +93,30 @@ func (r *OpenStackDataPlaneNodeSet) ValidateCreate() error {
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *OpenStackDataPlaneNodeSet) ValidateUpdate(old runtime.Object) error {
 	openstackdataplanenodesetlog.Info("validate update", "name", r.Name)
+	oldNodeSet, ok := old.(*OpenStackDataPlaneNodeSet)
+	if !ok {
+		return apierrors.NewInternalError(
+			fmt.Errorf("Expected a OpenStackDataPlaneNodeSet object, but got %T", oldNodeSet))
+	}
+
+	var errors field.ErrorList
+	// Changes to the baremetalSetTemplate after the initial deployment would necessitate
+	// a redeploy of the node. Thus we should block these changes and require the user to
+	// delete and redeploy should they wish to make such changes after the initial deploy.
+	if !reflect.DeepEqual(oldNodeSet.Spec.BaremetalSetTemplate, r.Spec.BaremetalSetTemplate) {
+		errors = append(errors, field.Forbidden(
```

Comment:
Well, this would not allow scale-out/scale-in as that's achieved updating `baremetalHosts` field.

Reply:
This is a great point. So yeah, let's change it to match what is happening in the `openstack-baremetal-operator` so that we don't block valid operations like scale out/in.

Reply:
So, as of the latest update to this PR, we will just return the error we get from the `openstack-baremetal-operator` `ValidateUpdate()` webhook. Which from the functional tests I have here will be:

```
<string>: admission webhook "vopenstackdataplanenodeset.kb.io" denied the request: OpenStackDataPlaneNodeSet.dataplane.openstack.org "edpm-compute-nodeset" is invalid: spec.baremetalSetTemplate: Forbidden: cannot change "bmhLabelSelector" nor "hardwareReqs" when previous count of "baremetalHosts" > 0
```